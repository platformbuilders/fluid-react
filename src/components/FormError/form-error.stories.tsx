import { faker } from '@faker-js/faker';
import { expect } from '@storybook/jest';
import type { Meta, StoryObj } from '@storybook/react';
import { within } from '@storybook/testing-library';
import FormError from './index';

//  Mocks
const mockErrorWords = faker.random.words(10);
const mockWords = faker.random.words(10);

const meta: Meta<typeof FormError> = {
  title: 'Components/FormError',
  component: FormError,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    error: {
      description: 'Error message to be displayed',
      type: 'string',
      control: { type: 'text' },
    },
    children: {
      description: 'Children to be displayed',
      type: 'string',
      control: { type: 'text' },
    },
  },
  args: {
    children: mockWords,
  },

  play: async ({ canvasElement, step }) => {
    const canvas = within(canvasElement);

    await step('FormError | Test Render', async () => {
      expect(canvas.getByText(mockWords)).toBeInTheDocument();
    });
  },
};

type Story = StoryObj<typeof FormError>;

export const Error: Story = {
  args: {
    error: mockErrorWords,
  },
};

export default meta;
