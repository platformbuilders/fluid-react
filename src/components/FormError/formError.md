
# FormError 

<!-- STORY -->

<br/>

### Source Code
<!-- SOURCE -->

<!-- PROPS -->

<!-- STORY HIDE START -->

content here won't be shown in stories

<!-- STORY HIDE END -->

