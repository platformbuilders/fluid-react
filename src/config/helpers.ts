import { DefaultTheme } from 'styled-components';
import { animations, colors, radius, spacing } from '../theme';

export const theme: DefaultTheme = {
  ...colors,
  ...animations,
  ...radius,
  ...spacing,
};
