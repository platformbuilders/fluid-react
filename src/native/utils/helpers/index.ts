export * from '@platformbuilders/helpers';
export * from './generateHaptic';
