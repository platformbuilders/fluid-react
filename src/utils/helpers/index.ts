export * from './haptic';
export { default as getTheme } from './getTheme';
export { default as ifStyle } from './ifStyle';
export { default as setStyleIOS } from './setStyleIOS';
export { default as switchStyle } from './switchStyle';
export { isIOS } from './isIOS';
