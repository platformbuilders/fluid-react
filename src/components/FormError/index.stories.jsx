import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text } from '@storybook/addon-knobs';
import markdown from './formError.md';
import FormError from './index';

storiesOf('FormError', module)
  .addDecorator(withKnobs)
  .addParameters({
    readme: {
      content: markdown,
    },
    viewport: {
      defaultViewport: '',
    },
  })
  .add('Example', () => (
    <FormError error={text('error', 'here is error message')} />
  ));
