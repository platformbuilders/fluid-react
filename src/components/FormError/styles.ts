import styled from 'styled-components/native';
import { getTheme } from '../../utils/helpers';

export const ErrorText = styled.Text`
  min-height: 17px;
  color: ${getTheme('failure')};
  margin-top: 5px;
`;
