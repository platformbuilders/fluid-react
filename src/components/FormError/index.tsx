import React, { FC, useContext } from 'react';
import { ThemeProvider } from 'styled-components';
import { ErrorText } from './styles';
import { ThemeContext } from '../ThemeContext';

type Props = {
  error: string | boolean | undefined;
  children?: JSX.Element | JSX.Element[];
  style?: object[];
};

const FormError: FC<Props> = ({ error = '', children, style }) => {
  const { theme } = useContext(ThemeContext);
  return (
    <ThemeProvider theme={theme}>
      <>
        {children}
        <ErrorText style={style}>{error}</ErrorText>
      </>
    </ThemeProvider>
  );
};

export default FormError;
