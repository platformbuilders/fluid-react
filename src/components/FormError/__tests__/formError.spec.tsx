import React from 'react';
import { shallow } from 'enzyme';
import faker from 'faker';
import { ThemeProvider } from 'styled-components';

import FormError, { Props } from '..';
import { theme } from '../../../config/helpers';

const defaultContent = 'Text';
const defaultProps: Props = {
  error: undefined,
};

describe('Component: FormError', () => {
  test('snapshots with default props', () => {
    const component = shallow(
      <ThemeProvider theme={theme}>
        <FormError {...defaultProps}>{defaultContent}</FormError>
      </ThemeProvider>,
    );

    expect(component).toMatchSnapshot();
  });

  test('snapshots with error props', () => {
    const component = shallow(
      <ThemeProvider theme={theme}>
        <FormError {...defaultProps} error="Error text">
          {defaultContent}
        </FormError>
      </ThemeProvider>,
    );

    expect(component).toMatchSnapshot();
  });

  test('should render the component when the condition is truthy', () => {
    const errorMock = faker.random.words(10);
    const component = shallow(
      <FormError {...defaultProps} error={errorMock}>
        {defaultContent}
      </FormError>,
    );

    expect(component.contains(defaultContent)).toEqual(true);
  });
});
