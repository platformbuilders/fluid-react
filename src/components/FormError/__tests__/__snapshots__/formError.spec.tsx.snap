// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: FormError snapshots with default props 1`] = `
<ContextProvider
  value={Object {}}
>
  <FormError>
    Text
  </FormError>
</ContextProvider>
`;

exports[`Component: FormError snapshots with error props 1`] = `
<ContextProvider
  value={Object {}}
>
  <FormError
    error="Error text"
  >
    Text
  </FormError>
</ContextProvider>
`;
